next.config.js

```diff
@@ -1,12 +1,18 @@
 const webpack = require('webpack');
 const pkg = require('./package.json');
+const withModuleTranspile = require('@weco/next-plugin-transpile-modules');
 const { BundleAnalyzerPlugin } = require('webpack-bundle-analyzer');
 const { findPages } = require('./docs/src/modules/utils/find');
 
 process.env.LIB_VERSION = pkg.version;
 
 module.exports = {
-  webpack: config => {
+  webpack: (config, options) => {
+    config = withModuleTranspile({
+      // transpile any peer dependent to @material-ui/core modules
+      transpileModules: ['material-ui-pickers'],
```

Comment:
Why is this needed? Can't find any documentation at `material-ui-pickers` that their code needs to be transpiled.

Reply:
The code needs to be transpired to make imports from @material-ui/core works. Cause material ui pickers use @material-ui/core as peer dependency

Reply:
I need to checkout the code and see this for myself. I don't understand why I need to transpile module A because it has a dependency on module B when B is already transpiled.

Reply:
Because module A (material ui core) is not a dependency of docs. And resolves only by webpack alias

Reply:
@eps1lon The discussion started in https://github.com/mui-org/material-ui/issues/13659#issuecomment-440624967.

Reply:
TL;DR: The purpose is not classic transpilation but abusing babel to implement webpack module resolve alias. 

Babel alias does only work for code that is actually piped through babel. Webpack alias works for every bundled package. So by piping `material-ui-pickers` again through babel we get the desired alias.

It's somewhat hacky but given `next`'s stance that webpack alias is the same as babel alias I don't think we have much of a choice. Additionally It just adds another potential "why is this happening" to the docs code and I much prefer being able to understand code without knowing the details of the build process.

I think it is technically solvable by putting all `package.json` in the `src/` directory which would remove the need for babel and wepack alias completely because yarn and lerna do all the work by module hoisting.

Reply:
Let's do that :)